staging/src/k8s.io/api/apps/v1/types.go

```diff
@@ -206,6 +206,29 @@ type StatefulSetStatus struct {
 	// newest ControllerRevision.
 	// +optional
 	CollisionCount *int32 `json:"collisionCount,omitempty" protobuf:"varint,9,opt,name=collisionCount"`
+
+	// Represents the latest available observations of a statefulset's current state.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
```

Comment:
Need `// +optional` tag for every `omitempty` field

https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#optional-vs-required